language/move-binary-format/src/compatibility.rs

```diff
@@ -268,3 +268,89 @@ fn type_parameter_phantom_decl_compatibile(
     // old_type_paramter.is_phantom => new_type_parameter.is_phantom
     !old_type_parameter.is_phantom || new_type_parameter.is_phantom
 }
+
+/// A simpler, and stricter compatibility checker relating to the inclusion of the old module in
+/// the new.
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq)]
+pub enum InclusionCheck {
+    Subset,
+    Equal,
+}
+
+impl InclusionCheck {
+    pub fn check(&self, old_module: &Module, new_module: &Module) -> PartialVMResult<()> {
+        let err = Err(PartialVMError::new(
+            StatusCode::BACKWARD_INCOMPATIBLE_MODULE_UPDATE,
+        ));
+
+        // Module checks
+        if old_module.address != new_module.address
+            || old_module.name != new_module.name
+            || old_module.file_format_version != new_module.file_format_version
```

Comment:
file format is an interesting one... is that strictly necessary? (If this is a requirement for the `Compatibility` checks then feel free to ignore, but I could imagine an additive update that only uses new aspects of the file format in newly added functions, that could be considered a superset of the old package).

Reply:
I was going back and forth on this and whether we should allow `old < new` or not care. I think relaxing this to `old < new` makes sense, and in the case of an `Equal` match we will enforce that the file format is the same.